test/framework/src/main/java/org/elasticsearch/test/ClusterServiceUtils.java

```diff
@@ -53,12 +53,12 @@ public static ClusterService createClusterService(ThreadPool threadPool, Discove
                 threadPool, () -> localNode);
         clusterService.setNodeConnectionsService(new NodeConnectionsService(Settings.EMPTY, null, null) {
             @Override
-            public void connectToNodes(Iterable<DiscoveryNode> discoveryNodes) {
+            public void connectToNodes(DiscoveryNodes discoveryNodes) {
                 // skip
             }
 
             @Override
-            public void disconnectFromNodesExcept(Iterable<DiscoveryNode> nodesToKeep) {
+            public void disconnectFromNodesExcept(DiscoveryNodes nodesToKeep) {
```

Comment:
why do we change these? Isn't Iterable more generic and would still accept DiscoveryNodes? also `disconnectFromNodesExcept` sounds like it would only be executed on a subset of and actual `DiscoveryNodes` instance. I think we should stick with what we had?